src/settings/autostartmanager/portalautostartmanager.cpp

```diff
@@ -0,0 +1,84 @@
+/*
+ *  Copyright © 2018-2022 Hennadii Chernyshchyk <genaloner@gmail.com>
+ *
+ *  This file is part of Crow Translate.
+ *
+ *  Crow Translate is free software; you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation; either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  Crow Translate is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a get of the GNU General Public License
+ *  along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#include "portalautostartmanager.h"
+
+#include "settings/appsettings.h"
+
+#include <QDBusReply>
+#include <QtCore>
+
+QDBusInterface PortalAutostartManager::s_interface(QStringLiteral("org.freedesktop.portal.Desktop"),
+                                                   QStringLiteral("/org/freedesktop/portal/desktop"),
+                                                   QStringLiteral("org.freedesktop.portal.Background"));
+
+PortalAutostartManager::PortalAutostartManager(QObject *parent)
+    : AbstractAutostartManager(parent)
+{
+}
+
+bool PortalAutostartManager::isAutostartEnabled() const
+{
+    return AppSettings().isAutostartEnabled();
+}
+
+void PortalAutostartManager::setAutostartEnabled(bool enabled)
+{
+    const QVariantMap options{
+        {QStringLiteral("reason"), QStringLiteral("Allow Crow Translate to manage autostart setting for itself.")},
+        {QStringLiteral("autostart"), enabled},
+        {QStringLiteral("commandline"), QStringList{QCoreApplication::applicationFilePath()}},
+        {QStringLiteral("dbus-activatable"), false},
+    };
+    const QDBusReply<QDBusObjectPath> reply = s_interface.call(QStringLiteral("RequestBackground"), QString(), options);
```

Comment:
I've left a TODO because while an empty string in place of `parent_window` works, I believe we shouldn't do it; instead, we should send an actual window id. It's doable, I just didn't find yet a suitable other Qt app where we can borrow a code snippet to get a window ID in XDP format from a QWindow.

Reply:
Oh, right: https://flatpak.github.io/xdg-desktop-portal/
I know how to fix it, let me do it.